src/main/scala/aserralle/akka/stream/kcl/scaladsl/KinesisWorkerSource.scala

```diff
@@ -40,10 +41,21 @@ object KinesisWorkerSource {
           val worker = workerBuilder(
             new IRecordProcessorFactory {
               override def createProcessor(): IRecordProcessor =
-                new IRecordProcessor(queue.offer,
-                                     settings.terminateStreamGracePeriod)
+                new IRecordProcessor(
+                  record =>
+                    (Exception.nonFatalCatch either Await.result(
+                      queue.offer(record),
+                      settings.backpressureTimeout) left)
+                      .foreach(
+                        err =>
+                          queue.fail(
+                            new RuntimeException("Back-pressure exhausted",
+                                                 err))),
```

Comment:
Would you rather create a custom Error in [Errors.scala](https://github.com/aserrallerios/kcl-akka-stream/blob/master/src/main/scala/aserralle/akka/stream/kcl/Errors.scala)?

Something like `case object BackpressureTimeout extends KinesisWorkerSourceError`?